include/pinocchio/multibody/joint/joint-ellipsoid.hpp

```diff
@@ -0,0 +1,601 @@
+//
+// Copyright (c) 2025 INRIA
+//
+
+#ifndef __pinocchio_multibody_joint_ellipsoid_hpp__
+#define __pinocchio_multibody_joint_ellipsoid_hpp__
+
+#include "pinocchio/macros.hpp"
+#include "pinocchio/multibody/joint/joint-base.hpp"
+#include "pinocchio/multibody/joint-motion-subspace.hpp"
+#include "pinocchio/math/sincos.hpp"
+#include "pinocchio/math/matrix.hpp"
+#include "pinocchio/spatial/inertia.hpp"
+#include "pinocchio/spatial/skew.hpp"
+#include "pinocchio/multibody/joint-motion-subspace.hpp"
+
+namespace pinocchio
+{
+  template<typename Scalar, int Options>
+  struct JointEllipsoidTpl;
+
+  template<typename _Scalar, int _Options>
+  struct traits<JointEllipsoidTpl<_Scalar, _Options>>
+  {
+    enum
+    {
+      NQ = 3,
+      NV = 3,
+      NVExtended = 3
+    };
+    typedef _Scalar Scalar;
+    enum
+    {
+      Options = _Options
+    };
+    typedef JointDataEllipsoidTpl<Scalar, Options> JointDataDerived;
+    typedef JointModelEllipsoidTpl<Scalar, Options> JointModelDerived;
+    typedef JointMotionSubspaceTpl<3, Scalar, Options, 3> Constraint_t;
+    typedef SE3Tpl<Scalar, Options> Transformation_t;
+
+    typedef MotionTpl<Scalar, Options> Motion_t;
+    typedef MotionTpl<Scalar, Options> Bias_t;
+
+    // [ABA]
+    typedef Eigen::Matrix<Scalar, 6, NV, Options> U_t;
+    typedef Eigen::Matrix<Scalar, NV, NV, Options> D_t;
+    typedef Eigen::Matrix<Scalar, 6, NV, Options> UD_t;
+
+    typedef Eigen::Matrix<Scalar, NQ, 1, Options> ConfigVector_t;
+    typedef Eigen::Matrix<Scalar, NV, 1, Options> TangentVector_t;
+
+    typedef boost::mpl::false_ is_mimicable_t; // not mimicable
+
+    PINOCCHIO_JOINT_DATA_BASE_ACCESSOR_DEFAULT_RETURN_TYPE
+  };
+
+  template<typename _Scalar, int _Options>
+  struct traits<JointDataEllipsoidTpl<_Scalar, _Options>>
+  {
+    typedef JointEllipsoidTpl<_Scalar, _Options> JointDerived;
+    typedef _Scalar Scalar;
+  };
+
+  template<typename _Scalar, int _Options>
+  struct traits<JointModelEllipsoidTpl<_Scalar, _Options>>
+  {
+    typedef JointEllipsoidTpl<_Scalar, _Options> JointDerived;
+    typedef _Scalar Scalar;
+  };
+
+  template<typename _Scalar, int _Options>
+  struct JointDataEllipsoidTpl : public JointDataBase<JointDataEllipsoidTpl<_Scalar, _Options>>
+  {
+    EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+    typedef JointEllipsoidTpl<_Scalar, _Options> JointDerived;
+    PINOCCHIO_JOINT_DATA_TYPEDEF_TEMPLATE(JointDerived);
+    PINOCCHIO_JOINT_DATA_BASE_DEFAULT_ACCESSOR
+
+    ConfigVector_t joint_q;
+    TangentVector_t joint_v;
+
+    Constraint_t S;
+    Constraint_t Sdot;
```

Comment:
You can leave the Sdot computation in the unit test to validate the c computation

Reply:
done.